eigsep_motor_control/encoder.py

```diff
@@ -27,70 +27,76 @@ def get_encoder(self, motor):
+                data = [int(d) for d in data.split()]
+                return np.array(data) / INT_LEN
+            except (serial.SerialException, ValueError) as e:
+                logging.error(f"Error reading analog data: {e}")
+                continue #try again
+
+        
     def read_volts(self, motor=None):
-        analog = self.read_analog()
-        if motor == "az":
-            return self.bit2volt(analog[0])
-        elif motor == "alt":
-            return self.bit2volt(analog[1])
-        else:
-            return self.bit2volt(analog)
-
+        while True:
+            try:
+                analog = self.read_analog()
+                if motor == "az":
+                    return self.bit2volt(analog[0], "az")
+                elif motor == "alt":
+                    return self.bit2volt(analog[1], "alt")
+                else:
+                    return [self.bit2volt(val, m) for val, m in zip(analog, ["az", "alt"])]
+            except Exception as e:
+                logging.error(f"Error converting analog to voltage for {motor}: {e}. Trying again.")
+                continue  # try again
+    
     def monitor(self, az_event, alt_event):
         names = ("az", "alt")
         events = (az_event, alt_event)
         vprev = None
         while True:
-            volts = self.read_volts()
-            msg = ""
-            for m, v in zip(names, volts):
-                msg += f"{m}: {v:.3f} V "
-            print(msg)
-            if vprev is None:
-                vprev = volts
-                continue
-            for i in range(2):
-                vmin = self.VOLT_RANGE[names[i]][0]
-                vmax = self.VOLT_RANGE[names[i]][1]
-                if volts[i] - vprev[i] > 0 and volts[i] >= vmax:
-                    logging.warning(f"Pot {names[i]} at max voltage.")
-                    events[i].set()
-                elif volts[i] - vprev[i] < 0 and volts[i] <= vmin:
-                    logging.warning(f"Pot {names[i]} at min voltage.")
-                    events[i].set()
-            vprev = volts
+            try:
+                volts = self.read_volts()
+                msg = ""
+                for m, v in zip(names, volts):
+                    msg += f"{m}: {v:.3f} V "
+                print(msg)
+                if vprev is None:
+                    vprev = volts
+                    continue
+                for i in range(2):
+                    vmin = self.VOLT_RANGE[names[i]][0]
+                    vmax = self.VOLT_RANGE[names[i]][1]
+                    if volts[i] - vprev[i] > 0 and volts[i] >= vmax:
+                        logging.warning(f"Pot {names[i]} at max voltage.")
+                        events[i].set()
+                    elif volts[i] - vprev[i] < 0 and volts[i] <= vmin:
+                        logging.warning(f"Pot {names[i]} at min voltage.")
+                        events[i].set()
+                    vprev = volts
+            except Exception as e:
```

Comment:
Not sure if this is necessary, since exceptions are caught in the other functions? Also the danger to excepting all Exceptions is that something bad might happen and we're just ignoring it -- for that reason I like to be explicit about what we're excepting